src/NuGet.Core/NuGet.PackageManagement/Projects/PackagesConfigNuGetProject.cs

```diff
@@ -60,11 +58,9 @@ public PackagesConfigNuGetProject(string folderPath, Dictionary<string, object>
                 throw new ArgumentNullException(nameof(folderPath));
             }
 
-            TargetFramework = GetMetadata<NuGetFramework>(NuGetProjectMetadataKeys.TargetFramework);
-
             PackagesConfigPath = Path.Combine(folderPath, "packages.config");
 
-            var projectName = GetMetadata<string>(NuGetProjectMetadataKeys.Name);
+            var projectName = Metadata[NuGetProjectMetadataKeys.Name];
```

Comment:
This breaks the functionality because you access the metadata before it's got initialized and it's inconsistent and confusing. At least it deserves a comment :)